File: src/main/java/fgo/cards/fgo/WorldsEndFlowerGarden.java
package fgo.cards.fgo;

import com.megacrit.cardcrawl.actions.common.ApplyPowerAction;
import com.megacrit.cardcrawl.characters.AbstractPlayer;
import com.megacrit.cardcrawl.monsters.AbstractMonster;

import fgo.cards.FGOCard;
import fgo.powers.WorldsEndFlowerGardenPower;

public class WorldsEndFlowerGarden extends FGOCard {
    public static final String ID = makeID(WorldsEndFlowerGarden.class.getSimpleName());

    public WorldsEndFlowerGarden() {
        super(ID, 1, CardType.POWER, CardTarget.SELF, CardRarity.UNCOMMON);
    }

    @Override
    public void use(AbstractPlayer p, AbstractMonster m) {
        addToBot(new ApplyPowerAction(p, p, new WorldsEndFlowerGardenPower(p, 10)));
    }
}

File: Scripts/Cards/WorldsEndFlowerGarden.cs
using Fgo.Scripts.Powers;
using Fgo.Scripts.Utils;
using MegaCrit.Sts2.Core.Commands;
using MegaCrit.Sts2.Core.Entities.Cards;
using MegaCrit.Sts2.Core.GameActions.Multiplayer;
using MegaCrit.Sts2.Core.HoverTips;
using MegaCrit.Sts2.Core.Localization.DynamicVars;
using STS2RitsuLib.Cards.DynamicVars;

namespace Fgo.Scripts.Cards;

public class WorldsEndFlowerGarden() : FgoCardModel(2, CardType.Power,
    CardRarity.Uncommon, TargetType.Self)
{
    protected override IEnumerable<IHoverTip> AdditionalHoverTips =>
    [
        HoverTipFactory.FromPower<WorldsEndFlowerGardenPower>(),
        FgoHoverTipHelper.CreateNpHoverTip()
    ];

    protected override IEnumerable<DynamicVar> CanonicalVars =>
    [
        ModCardVars.Power<WorldsEndFlowerGardenPower>(10)
    ];

    protected override void OnUpgrade()
    {
        DynamicVars[nameof(WorldsEndFlowerGardenPower)].UpgradeValueBy(5);
    }

    protected override async Task OnPlay(PlayerChoiceContext choiceContext, CardPlay play)
    {
        await PowerCmd.Apply<WorldsEndFlowerGardenPower>(choiceContext, Owner.Creature,
            DynamicVars[nameof(WorldsEndFlowerGardenPower)].BaseValue, Owner.Creature, this);
    }
}